Domain/Command{T}.cs

```diff
@@ -175,10 +175,21 @@ private bool CommandHandlerIsRegistered()
             }) != null;
         }
 
-        protected virtual void HandleCommandValidationFailure(TAggregate aggregate, ValidationReport validationReport)
+        protected virtual void HandleCommandValidationFailure(TTarget target, ValidationReport validationReport)
         {
-            ((dynamic) aggregate).HandleCommandValidationFailure((dynamic) this,
-                                                      validationReport);
+            if (target is IEventSourced)
+            {
+                ((dynamic) target).HandleCommandValidationFailure((dynamic) this,
+                                                                  validationReport);
+            }
+            else
```

Comment:
That seems like a reasonable future direction. I would change the type check to `EventSourcedAggregate` rather than `IEventSourced` though, and pull the `EventSourcedAggregate<T>.HandleCommandValidationFailure` up to `EventSourcedAggregate`.